src/poll.rs

```diff
@@ -56,14 +90,72 @@ fn emit_event(event_handler: &Mutex<dyn EventHandler>, res: Result<Event>) {
     }
 }
 
+impl PathData {
+    pub fn collect<BH: BuildHasher>(
+        path: &Path,
+        metadata: &Metadata,
+        build_hasher: Option<&BH>,
+        last_check: Instant,
+    ) -> Self {
+        let mtime = FileTime::from_last_modification_time(metadata).seconds();
+        let hash = metadata
+            .is_file()
+            .then(|| build_hasher.and_then(|bh| Self::hash_file(path, bh).ok()))
+            .flatten();
+        Self {
+            mtime,
+            hash,
+            last_check,
+        }
+    }
+
+    fn hash_file<P: AsRef<Path>, BH: BuildHasher>(path: P, build_hasher: &BH) -> io::Result<u64> {
+        let mut hasher = build_hasher.build_hasher();
+        let mut file = fs::File::open(path)?;
+        let mut buf = [0; 512];
```

Comment:
Normally I'd go with 1024 bytes, any reasing for the 512 ?

Reply:
The assumption is that if you're using the compare contents feature you probably expect these files to be quite small.  I debated between 256 and 512 even.  Also re BufferedReader, I understood it to actually be strictly worse than your own buffering like this when you expect to do a simple loop/copy.  I swore it was in BufferedReader's own docs but I'll double check and reevaluate

Reply:
Ok yeah makes sense. Re-reading the buf-reader docs it may be totally fair to not use it. (I'm pretty used to not using it being one of the top-reasons for people complaining about slow file read.)

If you would like to, you could test this with [walkdir](https://crates.io/crates/walkdir), hashing directories like your build folder or proc system and calculating the average + total time. Then you could compare it with/without bufreader or 256/512/1024 buffer size.